framework/console/controllers/BaseMigrateController.php

```diff
@@ -895,4 +909,22 @@ protected function generateMigrationSourceCode($params)
      * @param string $version migration version name.
      */
     abstract protected function removeMigrationHistory($version);
+
+    /**
+     * @param ReversibleMigrationInterface|Migration $migration
+     */
+    protected function applyChangesFrom(ReversibleMigrationInterface $migration)
```

Comment:
`applyChangesFrom` is too general with respect to that argument type hint. Its makes code less readable